### ISSUE TITLE
[Workaround] Failed to load shared library (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))

### ISSUE THREAD

**Issue by abetlen** [abetlen/llama-cpp-python#724]
This issue comes up on some Mac M series environments where cmake builds the llama.cpp shared library for the wrong architecture. The solution is to reinstall llama-cpp-python with:

```
CMAKE_ARGS="-DCMAKE_OSX_ARCHITECTURES=arm64" pip install --upgrade --verbose --force-reinstall --no-cache-dir llama-cpp-python
```

If anyone has a good way to automatically detect this in the CMakeLists.txt please let me know.

**Comment by timothyallan** [comment]
Yeah, I had to install the previous version. Good to know it wasn't just me!

**Comment by paschembri** [comment]
I had the bug until recently but I cannot reproduce with the 0.2.11 release. It seems resolved  ?

**Comment by steveoOn** [comment]
> This issue comes up on some Mac M series environments where cmake builds the llama.cpp shared library for the wrong architecture. The solution is to reinstall llama-cpp-python with:
> 
> ```
> CMAKE_ARGS="-DCMAKE_OSX_ARCHITECTURES=arm64" pip install --upgrade --verbose --force-reinstall --no-cache-dir llama-cpp-python
> ```
> 
> If anyone has a good way to automatically detect this in the CMakeLists.txt please let me know.

same issue, and use your cmake commander is worked.

**Comment by AhmadBaracat** [comment]
Using 0.2.11 as @timothyallan and @paschembri suggested worked for me

```
!CMAKE_ARGS="-DCMAKE_OSX_ARCHITECTURES=arm64" pip install --upgrade --verbose --force-reinstall --no-cache-dir llama-cpp-python==0.2.11
```

**Comment by kamransoomro84** [comment]
Thanks! Had the same issue. I wonder why this hasn't been updated in the installation docs.
